node_modules/oae-principals/tests/test-users.js

```diff
@@ -326,13 +327,74 @@ describe('Users', function() {
+            TenantsTestUtil.createTenantAndWait(globalAdminRestContext, tenantAlias, tenantAlias, tenantHost, {}, function(err) {
+                assert.ok(!err);
+
+                // Disable reCaptcha
+                ConfigTestUtil.updateConfigAndWait(globalAdminRestContext, tenantAlias, {'oae-principals/recaptcha/enabled': false}, function(err) {
+                    assert.ok(!err);
+
+                    var emailDomain1 = tenantHost;
+                    var emailDomain2 = TenantsTestUtil.generateTestTenantHost();
+
+                    // Verify accounts can be created for any email domain
+                    _verifyCreateUserWithEmailDomain(tenantHost, emailDomain1, true, function() {
+                        _verifyCreateUserWithEmailDomain(tenantHost, emailDomain2, true, function() {
+
+                            // Configure the tenant with an email domain
+                            RestAPI.Tenants.updateTenant(globalAdminRestContext, tenantAlias, {'emailDomain': emailDomain1}, function(err) {
+                                assert.ok(!err);
+
+                                // Only user accounts with an email that belongs to the configured email domain can be created
+                                _verifyCreateUserWithEmailDomain(tenantHost, emailDomain1, true, function() {
+                                    _verifyCreateUserWithEmailDomain(tenantHost, emailDomain2, false, function() {
+                                        return callback();
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Verify a user can or cannot be created with an email that belongs to a given email domain
+         *
+         * @param  {String}         tenantHost          The host of the tenant to create the user on
+         * @param  {String}         emailDomain         The email domain that the new user's email should belong to
+         * @param  {Boolean}        expectSuccess       Whether or not the user should be created
+         * @param  {Function}       callback            Standard callback function
+         * @api private
+         */
+        var _verifyCreateUserWithEmailDomain = function(tenantHost, emailDomain, expectSuccess, callback) {
+            var username = TestsUtil.generateTestUserId();
+            var email = TestsUtil.generateTestEmailAddress(null, emailDomain);
+            var anonymousRestContext = TestsUtil.createTenantRestContext(tenantHost);
+            RestAPI.User.createUser(anonymousRestContext, username, 'password', 'Test User', email, {'visibility': 'public'}, function(err) {
+                if (expectSuccess) {
+                    assert.ok(!err);
+                } else {
+                    assert.ok(err);
+                }
+                return callback();
+            });
+        };
```

Comment:
Can you extract `expectSuccess` into an `opts` object instead, so it's not just a cryptic boolean in the parameter list?

Also, it's possible we can just use a `assertCreateUserSucceeds` / `assertCreateUserFails` instead of defining a new function.